#161254663 create, list, update and delete articles

Eyansky opened the pull request:
#### What does this PR do?
Displays the following added functionalities:
- user can view articles
- user can create articles
- user can update articles
- user can delete articles

#### Description of Task to be completed?
- Articles form the backbone of our application and introduce us to CRUD which is the building block of most web applications.

#### How should this be manually tested?
- Clone this repo
```code
$ git clone https://github.com/andela/ah-orcas
```
- Navigate to this branch
```code
$ git checkout ft-create-articles-161254663
```
- Setup the environment variables, database and install the requirements. See sample.env
```code
$ source .env
```
- Start the server 
```code
$ python manage.py runserver
```
- Hit the endpoints created.
[![Run in Postman](https://run.pstmn.io/button.svg)](https://app.getpostman.com/run-collection/4da362e65fb40434f1d4)

#### Any background context you want to provide?
- I have rebased with Develop and as a result I found an existing folder called articles which I have successfully merged together with my work. I have however renamed the folder to article. All the work from articles folder has been integrated to the new folder article. 

#### What are the relevant pivotal tracker stories?
[161254663](https://www.pivotaltracker.com/story/show/161254663)

#### Screenshots (if appropriate)
- Get article endpoint screenshot
<img width="1078" alt="screen shot 2018-11-08 at 12 20 20" src="https://user-images.githubusercontent.com/26874561/48189282-e09bdd00-e350-11e8-874a-157d7570f3b1.png">

- Create article endpoint screenshot 
<img width="1036" alt="screen shot 2018-11-08 at 12 20 36" src="https://user-images.githubusercontent.com/26874561/48189323-f6a99d80-e350-11e8-9863-846ba92b0d56.png">

- Update article endpoint screenshot
<img width="1073" alt="screen shot 2018-11-08 at 12 20 45" src="https://user-images.githubusercontent.com/26874561/48189347-075a1380-e351-11e8-955c-3f2bc91ff323.png">

- Delete article endpoint screenshot
<img width="1058" alt="screen shot 2018-11-08 at 12 20 58" src="https://user-images.githubusercontent.com/26874561/48189368-12ad3f00-e351-11e8-9316-d4d5456637c4.png">

#### Questions:
- What do you think can be improved on?

kipropbrian commented:
Title of the PR is wrong.

code-sleuth commented on authors/apps/articles/models.py:
```diff
@@ -1,47 +0,0 @@
-from django.db import models
```
Why are you deleting everything in this file?

Eyansky replied:
I got rid of the articles folder and intergrated the changes to article folder. Thus rendering this file unwanted.

code-sleuth commented on authors/apps/articles/serializers.py:
```diff
@@ -1,22 +0,0 @@
-from rest_framework import serializers
```
Why are you deleting all this code?

Eyansky replied:
I got rid of the articles folder.

code-sleuth commented on authors/apps/articles/urls.py:
```diff
@@ -1,14 +0,0 @@
-from django.urls import path
```
Why are you deleting this code?

Eyansky replied:
I rebased from Develop and I had to sort out a merge conflict with Gidrafs work that was merged earlier for rate articles. The issue was I had a folder called article and he had articles. Soo we had to combine both to one folder. I therefore took his code and integrated it with mine in the article folder then got rid of his folder. That is why there is a-lot of deleted work. Both his work and mine are working perfectly after the conflict.

TheSteelGuy reviewed (commented):
Any particular reason why you are using a form to make requests   instead of JSON  format

Eyansky committed:
feat(articles) crud articles

- user can view articles
- user can create articles
- user can update articles
- user can delete articles
- fix hound errors
- fix travis build error
- fix merge conflict
- add tests

[Delivers #161254663]

Eyansky commented:
@TheSteelGuy I explained this  to you but for records, I used that because I needed postman to give me the functionality of adding a file so that I could add an image.

Eyansky commented:
@kipropbrian I have fixed the title Issue.

code-sleuth commented on authors/apps/article/models.py:
```diff
@@ -0,0 +1,104 @@
+
+class RateArticle(models.Model):
+    """
+    This is the article class. It holds data for the article.
```
Rate Article class

Eyansky replied:
What about it? I dont understand.

Eyansky replied:
This is gidrafs model files on rate articles. I merged them to my files when I rebased from develop  because his work was already merged.

Eyansky replied:
use ASK feedback.

code-sleuth commented on authors/apps/article/tests/test_article.py:
```diff
@@ -0,0 +1,101 @@
+from django.contrib.auth import get_user_model
+
+Article = apps.get_model('article', 'Article')
+# User = aps.get_model()
```
remove this line if its not needed.

Eyansky replied:
Done

TheSteelGuy commented on authors/apps/article/tests/test_rate_article.py:
```diff
@@ -53,21 +61,25 @@ def setUp(self):
-        self.rate_url = os.environ["URL"]+"api/article/"\
-            + self.slug + "/rate/"
-        self.view_rates_url = os.environ["URL"]+"api/article/rate/"
+        self.rate_url = os.environ["URL"] + \
```
Why use the environment variables to pass the URLs instead of building your URLs using Django inbuilt reverse mechanism. I don't think this is a good thing.

Eyansky replied:
This is Gidrafs work that has already been merged.  I just made some soft touches to it to make sure it worked together with my work. You should have raised this at Gidrafs pr.

TheSteelGuy replied:
👍

TheSteelGuy commented on authors/apps/articles/models.py:
```diff
@@ -1,47 +0,0 @@
-from django.db import models
-from django.core.validators import MaxValueValidator, MinValueValidator
```
Why are you deleting all these files?

Eyansky replied:
When i rebased from Develop, I found that Gidraf had created an articles folder and I had article folder without an "s" at the end. In this light I had to merge the work to one folder thus deleting the other folder and all its contents since they were all now intergrated to the article folder.

TheSteelGuy replied:
Make sense, was wondering what was happening, I remember this. That all for me.

evamaina commented on authors/apps/article/serializers.py:
```diff
@@ -0,0 +1,84 @@
+        except BaseException:
+            return {}
+
+    def update(self, instance, validated_data):
```
Maybe you should try to provide some Docstrings  in the future to describe what each  method is doing

Eyansky replied:
I had heavy comments all over my work which brought alot of linting issues thus i decided to scrap them all of and left only a-few. I will however explain everything during code review.
